docs/src/devdoc/architecture.rst

```diff
@@ -0,0 +1,80 @@
+.. _devdoc-architecture:
+
+Code organization
+=================
+
+The code is organized in three main modules, each in a separate directory:
+
+- ``equistore-core/`` contains the core library, implemented in Rust and exposed
+  to the outside world through a C API. This is also where the C++ API lives,
+  implemented as a header-only library in ``equistore.hpp``.
+- ``equistore/`` contains the Rust interface to equistore, using the C API
+  defined in ``equistore-core``, as well as the corresponding tests and examples
+- ``python/`` contains the Python interface to equistore, and the corresponding
+  tests and examples
+
+Finally, ``docs/`` contains the documentation for everything related to
+equistore.
+
+------------------------
+
+.. figure:: ../../static/images/API-organization.*
+    :width: 600px
+    :align: center
```

Comment:
Could you change the color of the arrows to gray like (0.5,0.5,0.5) in rgb. with this the diagramm also works in darkmode quite well.